src/Sign.Core/Tools/VsixSignTool/XmlSignatureBuilder.cs

```diff
@@ -87,19 +87,25 @@ private Stream CanonicalizeElement(XmlElement element, out string canonicalizati
         {
             //The canonicalization transformer can't reasonable do just an element. It
             //seems content to do an entire XmlDocument.
-            var transformer = new XmlDsigC14NTransform(false);
-            setCanonicalization?.Invoke(transformer.Algorithm);
 
-            var newDocument = new XmlDocument(_document.NameTable);
-            newDocument.LoadXml(element.OuterXml);
-            
-            transformer.LoadInput(newDocument);
+            var transformer = new XmlDsigC14NTransform(false);
+            string? algorithm = transformer.Algorithm;
```

Comment:
.NET 8 change `Transform.Algorithm`'s return type from `string` to `string?`, so this code raised a possible `null` error after updating to .NET 8.  A `null` or empty-length string is an invalid value for algorithm, so we should just fail fast if that's the case.